api/app/src/routes/medical/link.ts

```diff
@@ -59,38 +103,65 @@ router.delete(
  *
  * Builds and returns the current state of a patient's links across HIEs.
  *
- * @param   req.query.patientId Patient ID for which to retrieve links.
- * @return  {Link[]}            The patient's current and potential links.
+ * @param   req.query.patientId     Patient ID for which to retrieve links.
+ * @return  {PatientLinks}          The patient's current and potential links.
  */
 router.get(
   "/",
   asyncHandler(async (req: Request, res: Response) => {
     const cxId = getCxIdOrFail(req);
     const patientId = getPatientIdFromQueryOrFail(req);
 
-    // TODO: Here's some pseudocode for how we can do this for CommonWell
-    // (#374 will implement):
-
     const patient = await getPatient({ id: patientId, cxId });
-    console.log(patient);
+
     //  - initialize currentLinks, potentialLinks
+    const links: PatientLinks = {
+      currentLinks: [],
+      potentialLinks: [],
+    };
+
     //  - if the patient already has a link to a person
-    //      - need to verify link is still valid
-    //      - get the patient link -> CommonWell.getPatientLink()
-    //      - if the link exists and is >= LOLA 2
-    //        - add to currentLinks
-    //      - else
-    //        - this means the person was removed from CW
-    //        - TODO: ideas on how to handle this?
-    //  - if currentLinks is still empty
-    //      - initialize personResultsList
-    //      - if strong id is available
-    //        - add to personResultsList from strong ID search -> CommonWell.searchPerson()
+    if (
+      patient.linkData[LinkSource.commonWell]?.cw_link_id &&
+      patient.linkData[LinkSource.commonWell]?.cw_person_id
+    ) {
+      const { cw_person_id, cw_link_id } = patient.linkData[LinkSource.commonWell];
+      const personLink = await getLinkFromCommonwell(cw_person_id, cw_link_id);
+
+      if (personLink) {
+        links.currentLinks = [...links.currentLinks, personLink];
+      } else {
+        // - Remove from link data
+        await updatePatient({
+          id: patientId,
+          cxId,
+          linkData: {
+            ...patient.linkData,
+            [LinkSource.commonWell]: {},
+          },
+        });
+      }
+    }
+
+    if (!links.currentLinks.length) {
+      //      - initialize personResultsList
+      //      - if strong id is available
+      //        - add to personResultsList from strong ID search -> CommonWell.searchPerson()
+
+      const cwPatientId = `${patient.patientNumber}^^^urn:oid:${Config.getSystemRootOID()}`;
```

Comment:
shouldn't `Config.getSystemRootOID()` actually be the patient's org ID?

Reply:
AFAIK, yes.

Also, we have `urn:oid:` as a constant on `oid.ts`. And it would be good to have a small function for these kind of stuff, even if we're not planning to reuse it now - helps with keeping the logic easy to understand and allows for reusing and extensibility.

Reply:
Gotcha, i see Raf has a get patient with dependencies so i can use that to grab the proper org id in the meantime ill leave a TODO